governance/src/components/Staking/StakingModalForm.tsx

```diff
@@ -0,0 +1,306 @@
+  "1W",
+  "1M",
+  "3M",
+  "6M",
+  "1Y",
+  "4Y",
+  "custom",
+] as const;
+type DurationSelectOption = typeof durationSelectOptions[number];
+const durationSelectOptionsExcludeCustom = durationSelectOptions.filter(
+  (option) => option !== "custom"
+);
+
+const getDurationSelectOptionFromDuration = (
+  duration: Duration
+): DurationSelectOption => {
+  switch (duration.asDays()) {
+    case 7:
+      return "1W";
+    case 30:
+      return "1M";
+    case 91:
+      return "3M";
+    case 182:
+      return "6M";
+    case 365:
+      return "1Y";
+    case 4 * 365:
+      return "4Y";
+    default:
+      return "custom";
+  }
+};
+
+const LogoContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 64px;
+  height: 64px;
+  background: ${colors.red}1F;
+  border-radius: 100px;
+`;
+
+const StakingWarningHighlight = styled.strong`
+  color: ${colors.green};
+`;
+
+interface StakingModalFormProps {
+  initialStakingData: {
+    amount: BigNumber;
+    duration: Duration;
+  };
+  proceedToPreview: (amount: BigNumber, duration: Duration) => void;
+}
+
+const StakingModalForm: React.FC<StakingModalFormProps> = ({
+  initialStakingData,
+  proceedToPreview,
+}) => {
+  const { active } = useWeb3React();
+  const { data } = useRBNTokenAccount();
+
+  const [showCalendarPicker, setShowCalendarPicker] = useState(false);
+
+  const [amountInput, setAmountInput] = useState(
+    initialStakingData.amount.isZero()
+      ? ""
+      : formatUnits(initialStakingData.amount)
+  );
+  const [durationSelectValue, setDurationSelectValue] =
+    useState<DurationSelectOption>(
+      initialStakingData.duration.asDays() === 0
+        ? durationSelectOptions[0]
+        : getDurationSelectOptionFromDuration(initialStakingData.duration)
+    );
+  const [stakeDuration, setStakeDuration] = useState<Duration>(
+    initialStakingData.duration
+  );
+
+  const inputError = useMemo(() => {
+    console.log(data);
+    return Boolean(
+      amountInput &&
+        parseUnits(amountInput, 18).gt(
+          data ? data.walletBalance : BigNumber.from(0)
+        )
+    );
+  }, [amountInput, data]);
+
+  const canProceed = useMemo(() => {
+    return amountInput && !inputError && stakeDuration.asDays() >= 7;
+  }, [amountInput, inputError, stakeDuration]);
+
+  /**
+   * Update stake duration value
+   */
+  useEffect(() => {
+    switch (durationSelectValue) {
+      case "1W":
+        setStakeDuration(duration(7, "d"));
+        break;
+      case "1M":
+        setStakeDuration(duration(30, "d"));
+        break;
+      case "3M":
+        setStakeDuration(duration(91, "d"));
+        break;
+      case "6M":
+        setStakeDuration(duration(182, "d"));
+        break;
+      case "1Y":
+        setStakeDuration(duration(365, "d"));
+        break;
+      case "4Y":
+        setStakeDuration(duration(4 * 365, "d"));
+        break;
+      case "custom":
+        setStakeDuration(duration());
```

Comment:
This needs to throw when it is more than 2 years

Reply:
The `Duration` over here actually doesn't have any duration, and if the user close the calendar without updating `Duration` with some range, the input will then fallback to the multi-choice (1W, 1M, ...)